### PR TITLE
Adding info about new http client

### PR DESCRIPTION
**Pull request by MichaelGHSeg**
### Proposed changes

We're working on adding a new interface to customize the http client used in analytics-next node.

### Merge timing
~Once https://github.com/segmentio/analytics-next/pull/880 is merged~
Immediately

### Related issues (optional)
LIBWEB-1395

### CODE REVIEW

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 1]
```diff
@@ -553,7 +554,71 @@ Different parts of your application may require different types of batching, or
 const marketingAnalytics = new Analytics({ writeKey: 'MARKETING_WRITE_KEY' });
 const appAnalytics = new Analytics({ writeKey: 'APP_WRITE_KEY' });
 ```
+## AnalyticsHTTPClient
 
+We attempt to use the global `fetch` implementation if available in order to support several diverse environments.  Some special cases (e.g. http proxy) may require a different implementation for http communication.  A customized wrapper can be provided in the Analytics configuration to support this.  Here are a few approaches:
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/node/index.md:559]:
```suggestion
Segment attempts to use the global `fetch` implementation if available in order to support several diverse environments.  Some special cases (for example, http proxy) may require a different implementation for http communication.  You can provide a customized wrapper in the Analytics configuration to support this.  Here are a few approaches:
```

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 2]
```diff
@@ -284,6 +284,7 @@ Setting | Details
 `flushInterval` _number_ | The number of milliseconds to wait before flushing the queue automatically. The default is: `10000`
 `httpRequestTimeout` | The maximum number of milliseconds to wait for an http request. The default is: `10000`
 `disable` | Disable the analytics library for testing. The default is: `false`
+`httpClient` | Optional custom AnalyticsHTTPClient implementation to support alternate libraries or proxies.  Defaults to global fetch or node-fetch for older versions of node.
```
Comment (rchinn-segment) [inline src/connections/sources/catalog/libraries/server/node/index.md:287]:
```suggestion
`httpClient` *Optional* | A custom AnalyticsHTTPClient implementation to support alternate libraries or proxies.  Defaults to global fetch or node-fetch for older versions of node.
```

**Review by rchinn-segment — CHANGES_REQUESTED** [review]
@MichaelGHSeg looks good. Just a few updates to align this with our style guide.

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 3]
```diff
@@ -553,7 +554,71 @@ Different parts of your application may require different types of batching, or
...
+        auth: {
+          username: 'user',
+          password: 'pass',
+        },
+      },
+    ...options,
+  })
+}
+
+new Analytics({
```
Comment (silesky) [inline src/connections/sources/catalog/libraries/server/node/index.md:581]: missing "export const analytics = new Analytics..."

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 4]
```diff
@@ -553,7 +554,71 @@ Different parts of your application may require different types of batching, or
...
 const appAnalytics = new Analytics({ writeKey: 'APP_WRITE_KEY' });
 ```
+## AnalyticsHTTPClient
 
+Segment attempts to use the global `fetch` implementation if available in order to support several diverse environments.  Some special cases (for example, http proxy) may require a different implementation for http communication.  You can provide a customized wrapper in the Analytics configuration to support this.  Here are a few approaches:
+
+Use a custom fetch-like implementation with proxy (simple, recommended)
+```
+import { HTTPFetchFn } from '../lib/http-client'
+
```
Comment (silesky) [inline src/connections/sources/catalog/libraries/server/node/index.md:564]: Maybe also include `import axios from 'axios'`

File: src/connections/sources/catalog/libraries/server/node/index.md [hunk 5]
```diff
@@ -553,7 +554,71 @@ Different parts of your application may require different types of batching, or
 const marketingAnalytics = new Analytics({ writeKey: 'MARKETING_WRITE_KEY' });
 const appAnalytics = new Analytics({ writeKey: 'APP_WRITE_KEY' });
 ```
+## AnalyticsHTTPClient
 
+Segment attempts to use the global `fetch` implementation if available in order to support several diverse environments.  Some special cases (for example, http proxy) may require a different implementation for http communication.  You can provide a customized wrapper in the Analytics configuration to support this.  Here are a few approaches:
+
+Use a custom fetch-like implementation with proxy (simple, recommended)
+```
```
Comment (silesky) [inline src/connections/sources/catalog/libraries/server/node/index.md:562]: This should be a ```javascript

**Review by rchinn-segment — APPROVED** [review]

**Pull request merged**